### PR TITLE
Clone VMWare vm operation #529

### PR DESCRIPTION
**Pull request by mihaitusa**
- added CloneVM operation.
Signed-off-by: Mihail-Cristian Tusa <mihail-cristian.tusa@hpe.com>

### CODE REVIEW

File: score-vmware/src/main/java/io/cloudslang/content/vmware/services/VmService.java [hunk 1]
```diff
@@ -287,15 +291,177 @@
...
+    /**
+     * Method used to connect to a data center to clone an existing virtual machine.
+     *
+     * @param httpInputs Object that has all the inputs necessary to made a connection to data center
+     * @param vmInputs   Object that has all the specific inputs necessary to identify the virtual machine that will be cloned
+     * @return Map with String as key and value that contains returnCode of the operation, success message with task id
+     * of the execution or failure message and the exception if there is one
+     * @throws Exception
+     */
+    public Map<String, String> cloneVM(HttpInputs httpInputs, VmInputs vmInputs) throws Exception {
```
Comment (gituser173) [inline score-vmware/src/main/java/io/cloudslang/content/vmware/services/VmService.java]: I would move all this work inside a different class which handles only cloneVM related functionality as the VMService class is getting too big and is doing too much. Would do the same for the others also.

File: score-vmware/src/main/java/io/cloudslang/content/vmware/constants/Constants.java [hunk 2]
```diff
@@ -6,11 +6,14 @@
  */
 public class Constants {
     // VMware specific constants
```
Comment (gituser173) [inline score-vmware/src/main/java/io/cloudslang/content/vmware/constants/Constants.java]: maybe these constants can be moved into an enum? WDYT?

Comment (mihaitusa) [inline score-vmware/src/main/java/io/cloudslang/content/vmware/constants/Constants.java]: @gituser173  Yes... is a good approach.

File: score-vmware/src/main/java/io/cloudslang/content/vmware/services/VmService.java (hunk 1 above)
Comment (mihaitusa) [inline score-vmware/src/main/java/io/cloudslang/content/vmware/services/VmService.java]: @gituser173 @meshipeer  I agree with you... is getting bigger fast... but to put some executor class for each method within VmService will imply a major refactoring effort (with let say a factory pattern). I will do it if we have enough time...

**Pull request merged**